templates/go-carbon.conf.erb

```diff
@@ -24,38 +23,163 @@ aggregation-file = "<%= @whisper_aggregation_file %>"
 workers = <%= @whisper_workers %>
 # Limits the number of whisper update_many() calls per second. 0 - no limit
 max-updates-per-second = <%= @whisper_max_updates_per_second %>
-enabled = <%= @whisper_enabled %>
+# Softly limits the number of whsiper files that get created each second. 0 - no limit
+max-creates-per-second = <%= @whisper_max_creates_per_second %>
+# Make max-creates-per-second a hard limit. Extra new metrics are dropped. A hard throttle of 0 drops all new metrics.
+hard-max-creates-per-second = false
+# Sparse file creation
+sparse-create = false
+# use flock on every file call (ensures consistency if there are concurrent read/writes to the same file)
+flock = false
+# Use hashed filenames for tagged metrics instead of human readable
+# https://github.com/lomik/go-carbon/pull/225
+hash-filenames = true
 
 [cache]
 # Limit of in-memory stored points (not metrics)
 max-size = <%= @cache_max_size %>
 # Capacity of queue between receivers and cache
-input-buffer = <%= @cache_input_buffer %>
+# Strategy to persist metrics. Values: "max","sorted","noop"
+#   "max" - write metrics with most unwritten datapoints first
+#   "sorted" - sort by timestamp of first unwritten datapoint.
+#   "noop" - pick metrics to write in unspecified order,
+#            requires least CPU and improves cache responsiveness
+write-strategy = "<%= @cache_write_strategy %>"
 
 [udp]
 listen = "<%= @udp_listen %>"
 enabled = <%= @udp_enabled %>
 # Enable optional logging of incomplete messages (chunked by MTU)
 log-incomplete = <%= @udp_log_incomplete  %>
+# Optional internal queue between receiver and cache
+buffer-size = <%= @udp_buffer_size %>
 
 [tcp]
 listen = "<%= @tcp_listen %>"
 enabled = <%= @tcp_enabled %>
+# Optional internal queue between receiver and cache
+buffer-size = <%= @tcp_buffer_size %>
 
 [pickle]
 listen = "<%= @pickle_listen %>"
 enabled = <%= @pickle_enabled %>
 # Limit message size for prevent memory overflow
 max-message-size = <%= @pickle_max_message_size %>
+# Optional internal queue between receiver and cache
+buffer-size = <%= @pickle_buffer_size %>
 
 [carbonlink]
 listen = "<%= @carbonlink_listen %>"
 enabled = <%= @carbonlink_enabled %>
 # Close inactive connections after "read-timeout"
 read-timeout = "<%= @carbonlink_read_timeout %>"
-# Return empty result if cache not reply
-query-timeout = "<%= @carbonlink_query_timeout %>"
+
+# grpc api
+# protocol: https://github.com/lomik/go-carbon/blob/master/helper/carbonpb/carbon.proto
+# samples: https://github.com/lomik/go-carbon/tree/master/api/sample
+[grpc]
+listen = "127.0.0.1:7003"
+enabled = true
```

Comment:
Turning on grpc support?

Reply:
Also based off go-carbon.conf generated from the 0.13.0 deb install and current README. I'll add a parameter to disable this.